haskell.nix/library/.buildkite/pipeline.yml

```diff
@@ -0,0 +1,52 @@
+steps:
+
+- label: xrefcheck
+  commands:
+  - nix run github:serokell/xrefcheck
+
+- label: reuse
+  commands:
+  - nix build -L .#checks.x86_64-linux.reuse-lint
+
+- label: hlint
+  commands:
+  - nix build -L .#checks.x86_64-linux.hlint
+
+- label: stylish-haskell
+  commands:
+  - nix build -L .#checks.x86_64-linux.stylish-haskell
+
+- label: check-trailing-whitespace
+  commands:
+  - nix build -L .#checks.x86_64-linux.trailing-whitespace
+
+# - label: shellcheck
+#   commands:
+#   - nix build -L .#checks.x86_64-linux.shellcheck
+
+# build haskell components
+- group: build
+  steps:
+    - label: ghc{{matrix}}
+      command: nix build -L .#checks.x86_64-linux.ghc${{ matrix }}:build-all --keep-going
+
+      # list of GHC versions must be in sync with ghc-versions from flake.nix
+      matrix:
+        - "884"
+        - "8107"
+        - "902"
+
+# don't run tests until all components are built
+- wait
+
+# run the tests
+- group: test
+  steps:
+    - label: ghc{{matrix}}
+      command: nix build -L .#checks.x86_64-linux.ghc${{ matrix }}:test-all --keep-going
+
+      # list of GHC versions must be in sync with ghc-versions from flake.nix
```

Comment:
Since it's yaml, I think it should be possible to define matrix in only one place and reuse it here. Unless you want to demonstrate that the matrix for tests can be different, which sounds weird, I think we want it to be the same.

Reply:
thanks, updated